glibmm 2.80 bottle: link error when using Glib::DateTime::create_now_local()

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
% brew config
HOMEBREW_VERSION: 4.3.1-11-g7c62002
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c620029fb045e0f7cb6769f8ad542eb7f0bcaf7
Last commit: 4 hours ago
Core tap JSON: 20 May 19:19 UTC
Core cask tap JSON: 20 May 19:19 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.1 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.1/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.45.1 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false

% brew doctor
Your system is ready to brew.
```


### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I'm trying to link a program to bottled glibmm that uses the `Glib::DateTime::create_now_local()` method but I'm getting a linking error.

### What happened (include all command output)?

```
[1/1] Linking target program
FAILED: program 
c++  -o program program.p/program.cc.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-rpath,/opt/homebrew/Cellar/glibmm/2.80.0/lib -Wl,-rpath,/opt/homebrew/Cellar/glib/2.80.2/lib -Wl,-rpath,/opt/homebrew/opt/gettext/lib -Wl,-rpath,/opt/homebrew/Cellar/libsigc++/3.6.0/lib /opt/homebrew/Cellar/glibmm/2.80.0/lib/libglibmm-2.68.dylib /opt/homebrew/Cellar/glib/2.80.2/lib/libgobject-2.0.dylib /opt/homebrew/Cellar/glib/2.80.2/lib/libglib-2.0.dylib /opt/homebrew/opt/gettext/lib/libintl.dylib /opt/homebrew/Cellar/libsigc++/3.6.0/lib/libsigc-3.0.dylib
Undefined symbols for architecture arm64:
  "Glib::DateTime::create_now_local(long long)", referenced from:
      _main in program.cc.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

### What did you expect to happen?

The application should link without errors.

The error seem to happen only when the bottled version of glibmm is used and disappears if one re-builds glibmm from source (e.g. with `brew reinstall -s glibmm`). Maybe the bottle needs to be re-built?

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install --force-bottle glibmm

mkdir glibmm-repro
cd glibmm-repro

cat >meson.build <<EOF
project('program', 'cpp', default_options: ['cpp_std=c++20'])
glibmm_dep = dependency('glibmm-2.68')
program_name = 'program'
cpp_sources = ['program.cc']
executable(program_name,
  cpp_sources,
  dependencies: [glibmm_dep]
)
EOF

cat >program.cc <<EOF
#include <glibmm.h>

int main() {
    Glib::DateTime::create_now_local(0);
    return 0;
}
EOF

meson setup build
meson compile -C build
```

--- a/Formula/g/glibmm.rb
+++ b/Formula/g/glibmm.rb
@@ -4,6 +4,7 @@ class Glibmm < Formula
   url "https://download.gnome.org/sources/glibmm/2.80/glibmm-2.80.0.tar.xz"
   sha256 "539b0a29e15a96676c4f0594541250566c5ca44da5d4d87a3732fa2d07909e4a"
   license "LGPL-2.1-or-later"
+  revision 1
 
   bottle do
     sha256 cellar: :any, arm64_sonoma:   "8146911c8de1e567e11c9125e200d980aa8fcc949670be77f005337fc13d1539"
@@ -27,6 +28,12 @@ class Glibmm < Formula
     system "meson", "setup", "build", "-Dbuild-examples=false", *std_meson_args
     system "meson", "compile", "-C", "build", "--verbose"
     system "meson", "install", "-C", "build"
+
+    return unless OS.mac?
+
+    inreplace lib/"glibmm-2.68/proc/gmmproc",
+              "#{HOMEBREW_LIBRARY}/Homebrew/shims/mac/super/m4",
+              "#{HOMEBREW_PREFIX}/bin/m4"
   end
 
   test do

--- a/Formula/g/glibmm.rb
+++ b/Formula/g/glibmm.rb
@@ -7,13 +7,13 @@ class Glibmm < Formula
   revision 1
 
   bottle do
-    sha256 cellar: :any, arm64_sonoma:   "8146911c8de1e567e11c9125e200d980aa8fcc949670be77f005337fc13d1539"
-    sha256 cellar: :any, arm64_ventura:  "80d92647ef150c09cb50379b97a7ff74ce525266a9d43b0b72d094642c688d0a"
-    sha256 cellar: :any, arm64_monterey: "a59f817889d8bff3abe041194aa52717d28e9429e069005e8f92d57805fe0db1"
-    sha256 cellar: :any, sonoma:         "349305c1bf11e5833b7288e1898f63ea520bb259eec32a957fceda0a59b60186"
-    sha256 cellar: :any, ventura:        "6ec8c02168f12f069a8d75cfa6f81af8fd906b1ffb1f8678a865b6879adc28c4"
-    sha256 cellar: :any, monterey:       "9bfc78450bef875c2a85fc9ca85f6036914a4d56ba9fcd0139404f5308c14c97"
-    sha256               x86_64_linux:   "dfa186baebe5dca5af93f3ca2799e34ad128efedc6183bdad0f3e73bbdf41acd"
+    sha256 cellar: :any, arm64_sonoma:   "874cd180abccbe1e4637ba6459ee2c4aa60b65adc240387ae6aa3a47ac040e11"
+    sha256 cellar: :any, arm64_ventura:  "eb5880ccddc02dd49653b67767b06d9aa6748e2895ebf6a7adcff17651063a2a"
+    sha256 cellar: :any, arm64_monterey: "0a8b3f36488bd9bda5ea87ef2123e905321212397c690c1c21f26b19e47496a1"
+    sha256 cellar: :any, sonoma:         "318957fb8c9cc96ed11a7df991c065dfb0b3409a3f3363addbac27ec12ee290b"
+    sha256 cellar: :any, ventura:        "94875f1cf28156ff0fd0a5b71ba8282ac62ac146db0e69c943246f8df2f15d0f"
+    sha256 cellar: :any, monterey:       "65cc335fd272f9dc8f3cc40d6ab66de7a7220a6a5a370474356025b18b7b4e52"
+    sha256               x86_64_linux:   "10ff3f89ee53bcd2daaee31764896d968e9292ba7cdd46cde031c658fed1b2dc"
   end
 
   depends_on "meson" => :build

